
R version 4.5.0 (2025-04-11) -- "How About a Twenty-Six"
Copyright (C) 2025 The R Foundation for Statistical Computing
Platform: x86_64-pc-linux-gnu

R is free software and comes with ABSOLUTELY NO WARRANTY.
You are welcome to redistribute it under certain conditions.
Type 'license()' or 'licence()' for distribution details.

R is a collaborative project with many contributors.
Type 'contributors()' for more information and
'citation()' on how to cite R or R packages in publications.

Type 'demo()' for some demos, 'help()' for on-line help, or
'help.start()' for an HTML browser interface to help.
Type 'q()' to quit R.

> # This file is part of the standard setup for testthat.
> # It is recommended that you do not modify it.
> #
> # Where should you do additional test configuration?
> # Learn more about the roles of various files in:
> # * https://r-pkgs.org/tests.html
> # * https://testthat.r-lib.org/reference/test_package.html#special-files
> 
> library(testthat)
> library(RMariaDB)
> 
> test_check("RMariaDB")
Could not initialise default MariaDB database. If MariaDB is running
check that you have a ~/.my.cnf file that contains a [rs-dbi] section
describing how to connect to a test database.
[ FAIL 1 | WARN 0 | SKIP 28 | PASS 0 ]

══ Skipped tests (28) ══════════════════════════════════════════════════════════
• mariadbHasDefault() is not TRUE (1):
  'test-dbQuoteIdentifier_MariaDBConnection_Id.R:2:3'
• On CRAN (3): 'test-astyle.R:2:3', 'test-queries.R:73:3',
  'test-timezone.R:1:1'
• Test database not available (24): 'test-dbConnect.R:2:3',
  'test-dbConnect.R:10:3', 'test-dbConnect.R:18:3',
  'test-dbQuoteIdentifier.R:2:3', 'test-dbQuoteIdentifier.R:12:3',
  'test-dbQuoteIdentifier.R:22:3', 'test-dbQuoteIdentifier.R:32:3',
  'test-dbQuoteIdentifier.R:42:3', 'test-dbQuoteIdentifier.R:52:3',
  'test-dbQuoteIdentifier.R:59:3', 'test-dbQuoteIdentifier.R:79:3',
  'test-dbQuoteIdentifier.R:93:3', 'test-dbWriteTable.R:12:3',
  'test-dbWriteTable.R:23:3', 'test-dbWriteTable.R:34:3',
  'test-dbWriteTable.R:43:3', 'test-dbWriteTable.R:62:3',
  'test-dbWriteTable.R:72:3', 'test-dbWriteTable.R:82:3', 'test-queries.R:3:3',
  'test-queries.R:14:3', 'test-queries.R:23:3', 'test-queries.R:48:3',
  'test-queries.R:85:3'

══ Failed tests ════════════════════════════════════════════════════════════════
── Error ('test-DBItest.R:4:3'): (code run outside of `test_that()`) ───────────
Error: Failed to connect: Can't connect to local MySQL server through socket '/var/run/mysqld/mysqld.sock' (2)
Backtrace:
     ▆
  1. ├─DBItest::test_all() at test-DBItest.R:4:3
  2. │ └─DBItest::test_getting_started(...) at DBItest/R/test-all.R:29:3
  3. │   └─DBItest:::run_tests(...) at DBItest/R/test-getting-started.R:20:3
  4. │     └─DBItest:::local_connection(ctx) at DBItest/R/run.R:33:5
  5. │       └─DBItest:::connect(ctx, ...) at DBItest/R/utils.R:13:3
  6. │         └─rlang::eval_tidy(quo(dbConnect(ctx$cnr, !!!quos))) at DBItest/R/context.R:106:3
  7. ├─DBI::dbConnect(ctx$cnr)
  8. └─DBI::dbConnect(ctx$cnr)
  9.   ├─base::do.call(dbConnect, all_args)
 10.   ├─DBI (local) `<stndrdGn>`(drv = `<MrDBDrvr>`, dbname = "test", load_data_local_infile = FALSE)
 11.   └─RMariaDB (local) `<stndrdGn>`(drv = `<MrDBDrvr>`, dbname = "test", load_data_local_infile = FALSE)
 12.     └─RMariaDB (local) .local(drv, ...)
 13.       └─RMariaDB:::connection_create(...)

[ FAIL 1 | WARN 0 | SKIP 28 | PASS 0 ]
Error: Test failures
Execution halted
